### DISCUSSION TITLE
Custom errors and disabling logging for .should()

### DISCUSSION THREAD

**Discussion by SebMoreno** [cypress-io/cypress#29479] [Questions and Help]
Is there a way to:
1. have a custom error displayed for an assertion made with .should()?
2. have a custom error be displayed without the original chai/cypress error?

I've tried several things as shown below but I haven't really been able to fully replace nor turn off an error message made with .should()

```
cy.visit("https://www.google.com/en");

// error message: expected <input.gNO89b> to have value Google Search
cy.get("input[type=submit]").eq(2).should("have.value", "Google Search");

// error message is the same as the original, does nothing
cy.get("input[type=submit]").eq(2).should("have.value", "Google Search", "Custom error"); 

// error message: Custom error: expected <input.gNO89b> to have value Google Search
// this displays the custom error but appends the original chai/cypress
cy.get("input[type=submit]").eq(2).should(btn => {
    expect(btn, "Custom error").to.have.value("Google Search");
}); 
```

**Comment by portyu9** [comment]
Hi @SebMoreno 

For a completely custom failure message, use the callback form of `.should()` and throw your own error instead of using a Chai assertion:

```js
cy.get('input[type=submit]').eq(2).should(($input) => {
  if ($input.val() !== 'Google Search') {
    throw new Error('Custom error')
  }
})
```

Cypress will still retry the `.should()` callback until it passes or times out, but the final failure will use your custom error rather than Chai's generated assertion message.

Also, `.should()` itself does not have a documented `{ log: false }` option. Commands that support `log: false` can suppress their own Command Log entries, but that option does not apply to the assertion.

Please mark this answer as accepted if it helped, thank you.
